### PR TITLE
Lookup joins can use partial indexes

### LINKED ISSUE #8516: release-20.2: opt: generate lookup joins on partial indexes
PR: https://github.com/cockroachdb/cockroach/pull/54362

From release notes:
> The optimizer can now use [partial indexes](../v20.2/partial-indexes.html) for [lookup join](../v20.2/joins.html#lookup-joins) operations. This results in potentially more efficient query plans for joins on tables with partial indexes. [#54362][#54362] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #8516.

### CODE REVIEW

File: v20.2/joins.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ Hash joins are performed on two tables as follows:
 
 ### Lookup joins
 
-The [cost-based optimizer](cost-based-optimizer.html) decides when it would be beneficial to use a lookup join. Lookup joins are used when there is a large imbalance in size between the two tables, as it only reads the smaller table and then looks up matches in the larger table. A lookup join requires that the right-hand (i.e., larger) table is indexed on the equality column.
+The [cost-based optimizer](cost-based-optimizer.html) decides when it would be beneficial to use a lookup join. Lookup joins are used when there is a large imbalance in size between the two tables, as it only reads the smaller table and then looks up matches in the larger table. A lookup join requires that the right-hand (i.e., larger) table is indexed on the equality column, or [partially indexed](partial-indexes.html) on the subset of rows in the equality column.
```
Comment (mgartner) [inline v20.2/joins.md:134]: [nit] `on the subset of rows in the equality column` is confusing to me. Maybe something like:

> A lookup join requires that the right-hand (i.e., larger) table is indexed on the equality column. A [partial index](partial-indexes.html) can only be used if it contains the subset of rows being looked-up.

**Review by mgartner — APPROVED** [review]
LGTM!

**Review by mgartner — APPROVED** [review]
💯

**Review by lnhsingh — COMMENTED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8644#-:-MJh52JiCcMbJp2dpY03:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @lnhsingh and @mgartner)

**Pull request merged**